{"cell_type":{"c3489c02":"code","6dda7b7c":"code","deae29e8":"code","8a2b8edb":"markdown","5bfb98bc":"markdown"},"source":{"c3489c02":"import requests\nfrom datetime import datetime\n\nimport pandas as pd","6dda7b7c":"API_BASE = 'https:\/\/api.binance.com\/api\/v3\/'\n\nLABELS = [\n    'open_time',\n    'open',\n    'high',\n    'low',\n    'close',\n    'volume',\n    'close_time',\n    'quote_asset_volume',\n    'number_of_trades',\n    'taker_buy_base_asset_volume',\n    'taker_buy_quote_asset_volume',\n    'ignore'\n]","deae29e8":"params = {\n    'symbol': 'BTCUSDT',\n    'interval': '1m',\n    'startTime': int(datetime(2019, 3, 12).timestamp() * 1000),\n    'limit': 1440\n}\n\nresponse = requests.get(f'{API_BASE}klines', params)\ndf = pd.DataFrame(response.json(), columns=LABELS)\ndf.index = df['open_time'].apply(lambda x: datetime.fromtimestamp(x \/ 1000))\ndf.drop(['open_time', 'close_time'], axis=1).apply(pd.to_numeric).plot()","8a2b8edb":"Gap confirmed.","5bfb98bc":"A quick confirmation of the fact that the missing data in the dataset is indeed due to the response given by Binance's API and not an error in the data processing. Reason being [this post by @dhhlys](https:\/\/www.kaggle.com\/jorijnsmit\/binance-full-history\/discussion\/134390#999388):\n> Hi, thanks for the sharing. I have downloaded btcusd file. I found some data is missing. For example, there is no minute data from 20190312_02:00:00 to 08:00:00. Can you help to check it?"}}